templates/CRM/Contribute/Form/Contribution/MainMembershipBlock.tpl

```diff
@@ -134,7 +134,7 @@
       var priceSetName = "price_"+{/literal}'{$membershipFieldID}'{literal};
       var considerUserInput = {/literal}'{$takeUserSubmittedAutoRenew}'{literal};
       if ( memTypeId ) considerUserInput = false;
-      if ( !memTypeId ) memTypeId = cj('input:radio[name='+priceSetName+']:checked').attr('membership-type');
+      if ( !memTypeId ) memTypeId = cj('input:radio[name='+priceSetName+']:checked').attr('data-membership-type-id');
```

Comment:
@patricklam if this works I think it would be sligtly cleaner 
```
      if ( !memTypeId ) memTypeId = cj('input:radio[name='+priceSetName+']:checked').data('membership-type-id');
```

I'll create a PR for ^^ against the rc (5.70) & if you confirm it works we can merge it - otherwise we need to move this from master to the rc since I believe it is a regression

Reply:
I can confirm that .data('membership-type-id') works. I'm not sure how long this has been broken. Thanks!

Reply:
thanks @patricklam  - can you comment that it works over on https://github.com/civicrm/civicrm-core/pull/29118 - so I can get someone to merge it